explainer.md

```diff
@@ -1,624 +1,397 @@
 
-i. When a pseudo element is created for shared elements in the old page in step
-1, the pseudo-class "transition" is enabled for each shared element.
+Multiple elements could have the same `pagetransitiontag` value, which is useful for aggregate content such as comments.
 
-ii. In step 3 when saving state from the old page, the completed ComputedStyle
-on the pseudo elements is saved in addition to the computed size and transform.
+Items on Page-A and Page-B which have the same `pagetransitiontag` are automatically associated with each other. If multiple items have the same value, they're associated in DOM order.
 
-iii. In step 7 when applying a UA stylesheet to pseudo elements on the new page,
-the saved style is also applied to the old pseudo elements :
+### CSS properties
 
 ```css
-::shared-container(header-id) {
-  /* This is the saved output referenced in step 3. Applied before updating to
-   values from the shared element in the new DOM. */
-  width: 100px;
-  height: 100px;
-  transform: translate(8px, 308px);
-  border: 10px solid black;
-  border-radius: 10% 10%;
-  box-shadow: 0px 0px 10px;
+.header {
+  page-transition-tag: header;
+  page-transition-capture: container-and-child;
 }
+```
 
-::shared-old(header-id) {
-  position: absolute;
-  width: 100%;
-  height: 100%;
+This operates similar to the attributes, but it's easier to give multiple elements the same tag. It's also easier to change values depending on viewport (`@media`) and browser support (`@supports`).
 
-  /* This is the saved output referenced in step 3. */
-  content: cached-element(header-id);
-  top: -10px;
-  left: -10px;
-}
+### JS API
+
+```js
+// In Page-A
+addEventListener('navigate', (event) => {
+  event.prepareSameOriginPageTransition((transition) => {
+    transition.offerItem('header', document.querySelector('.header'), {
+      data: {…},
+      capture: 'container-and-child',
+    });
+  });
+});
 ```
 
-An example simulating the steps above using the existing `element()` function
-is [here](https://jsbin.com/vesokanumu/edit?html,output) (open in Firefox).
+Expanding on the capabilities of the CSS properties, a JS API allows the developer to change which parts are offered depending on the destination of the navigation, and the direction of navigation (back vs forward). This builds on top of the [app-history API](https://github.com/WICG/app-history).
 
-[![Video Link for Shared Element Transition](https://img.youtube.com/vi/SGnZN3NE0jA/0.jpg)](https://youtu.be/SGnZN3NE0jA)
+Also, `data` can be associated with the element. This can be anything structured-clonable, and will be made available to Page-B.
 
-## Modifications to element()
+```js
+// In Page-B
+document.performTransition((transition) => {
+  const pageAHeader = transition.offeredItems.get('header');
+  console.log(pageAHeader.data);
+  const pageBHeader = transition.createItem(document.querySelector('.header'), {
```

Comment:
Yeah, you'd have to remove them at the end of the transition. And fair point that the JS API can be limited to the particular transition.

But I think this ties into my other comment about not coupling with app-history. If you wanted that to be "tied" to a particular navigation then the offered elements have to be associated to a navigation event somehow. And I didn't want to add a navigation event to the documentTransition API since app-history is doing that already. But also not tightly couple with it. That's why I was hoping the semantics to be "this element is offered(old page)/used(new page) for whatever the next transition is" from the documentTransition API perspective. And developers relying on other events to set it based on the particular situation.

Maybe we can leave this as an open question and come back to it when we narrow down API details?